onnxscript/ir/_core.py

```diff
@@ -2961,7 +2971,12 @@ def outputs(self) -> MutableSequence[Value]:
     def attributes(self) -> OrderedDict[str, Attr]:
         return self._attributes
 
-    def __getitem__(self, index: int) -> Node:
+    @typing.overload
+    def __getitem__(self, index: int) -> Node: ...
+    @typing.overload
+    def __getitem__(self, index: slice) -> tuple[Node, ...]: ...
```

Comment:
nit

```suggestion
    def __getitem__(self, index: slice) -> Sequence[Node]: ...
```

here and elsewhere, thanks!